sdk/core/azure-core/tests/test_universal_pipeline.py

```diff
@@ -408,3 +414,72 @@ def emit(self, record):
     assert mock_handler.messages[4].message == 'Response headers:'
 
     mock_handler.reset()
+
+def test_retry_seekable_stream():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self._first = True
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            if self._first:
+                self._first = False
+                request.body.seek(0,2)
+                raise AzureError('fail on first')
+            position = request.body.tell()
+            assert not position
+            return HttpResponse(request, None)
+
+    data = BytesIO(b"Lots of dataaaa")
+    http_request = HttpRequest('GET', 'http://127.0.0.1/')
+    http_request.set_streamed_data_body(data)
+    http_retry = RetryPolicy(retry_total = 1)
+    pipeline = Pipeline(MockTransport(), [http_retry])
+    pipeline.run(http_request)
+
+def test_retry_seekable_file():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self._first = True
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            if self._first:
+                self._first = False
+                for value in request.files.values():
+                    name, body = value[0], value[1]
+                    if name and body and hasattr(body, 'read'):
+                        body.seek(0,2)
+                        raise AzureError('fail on first')
+            for value in request.files.values():
+                name, body = value[0], value[1]
+                if name and body and hasattr(body, 'read'):
+                    position = body.tell()
+                    assert not position
+                    return HttpResponse(request, None)
+
+    file_name = 'test_retry_seekable_file'
```

Comment:
Detail: usually it's better to do a contextmanager with tempfile.NamedTemporaryFile, but that's not critical here

Reply:
I was thinking of suggesting this but the docs state *"Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms"* and it seems like that might apply here?

Reply:
Well, I guess you can still `close` first inside

Reply:
Hum, I get your point, guess it's not worth the risk of having a test that could fail randomly on different plateform. I could argue we could do a tmpdir then :p 
Usually, I don't like temp resourses locally in the test folder, that the best way to have test that behave weird, or unexpected files on Github.

Reply:
Oh, definitely agree. I think actually if you call `close` first in the context manager (which I just didn't think of at first, unusual outside this context) then it would be fine.